crates/script/src/receipts.rs

```diff
@@ -39,6 +39,7 @@ pub async fn check_tx_status(
             return Ok(receipt.into());
         }
 
+        let mut retries = 0;
```

Comment:
can we use the retry utility here

Reply:
I am not sure we can since `retries` within `Retry` utility applies for all errors (while in this case we want to apply retry only for fatal errors while for timeout error we want to loop until receipt is available). does this make sense?

Reply:
That's what the "until_break" variant is for. I would prefer using the Retry struct because it also emits a warning when retrying.

Reply:
@DaniPopes pls check https://github.com/foundry-rs/foundry/pull/9650/commits/c35a798669670848853aac05ee9d29dbab7f085e with `Continue` variant. I have tested it on base sepolia with 100 txes but granted it doesn't cover all potential scenarios. One thing to notice is that progress bar becomes flaky now when a retry as `Retry` utility prints warning